Tests/OpenAPIRuntimeTests/URICoder/Encoding/Test_URIEncoder.swift

```diff
@@ -23,4 +23,12 @@ final class Test_URIEncoder: Test_Runtime {
         let encodedString = try encoder.encode(Foo(bar: "hello world"), forKey: "root")
         XCTAssertEqual(encodedString, "bar=hello+world")
     }
+    
+    func testNestedEncoding() throws {
+        struct Foo: Encodable { var bar: String }
+        let serializer = URISerializer(configuration: .deepObjectExplode)
+        let encoder = URIEncoder(serializer: serializer)
+        let encodedString = try encoder.encode(Foo(bar: "hello world"), forKey: "root")
+        XCTAssertEqual(encodedString, "root%5Bbar%5D=hello+world")
```

Comment:
Are we sure that deepObject uses `+` instead of `%20` for encoding spaces? Can you find it in the RFC?

Reply:
Nice catch! Unfortunately RFC doesn't refer anything regarding deepObject. My reading of OpenAPISpec[[1]](https://spec.openapis.org/oas/v3.1.0.html#style-values) is that deepObject uses form parameters configuration, so I think we should use `%20` encoding for spaces.

## 
 1. https://spec.openapis.org/oas/v3.1.0.html#style-values

Reply:
Yeah that makes sense, thanks for checking.